### PR TITLE
Added license notes to 3 books and added type of book (pdf) for one 

### PR DESCRIPTION
**Pull request by manvith12**
## What does this PR do?
Add info based on #11355

#### Description
This PR adds license notes to the following books located at books:

1. **A Complete Guide to Standard C++ Algorithms**  
   License Line found on Page 4 of the PDF.  
   "This work may be distributed and/or modified under the conditions of the CC-BY-NC-SA license."  
   Added (CC-BY-NC-SA)

2. **C++ GUI Programming with Qt 3**  
   License Line found on Page 4 of the PDF.  
   "This material may only be distributed subject to the terms and conditions set forth in the Open Publication License, v1.0 or later."  
   Added (OPl)

3. **C++ Notes for Professionals**  
   License Line found on Page 19 of the PDF.  
   "The content is written by the beautiful people at Stack Overflow. Text content is released under Creative Commons BY-SA."  
   Added (CC BY-SA)

this pr also added info on type of book :

1. **C++ Language**
    checked the  book and added (HTML)

#### Why is this valuable (or not)?
This information clarifies the licensing for these resources, ensuring users understand the terms under which they can use, modify, and distribute the content.

#### How do we know it's really free?
The licenses clearly state that the materials can be used freely within the specified conditions, which aligns with the principles of open educational resources.

#### For book lists, is it a book? For course lists, is it a course? etc.
It's a book.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up
- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by manvith12** [comment]
i dont understand whats the lint error , could anyone please explain

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -512,15 +512,15 @@ Books on general-purpose programming that don't focus on a specific language are
 
 ### <a id="cpp"></a>C++
 
-* [A Complete Guide to Standard C++ Algorithms](https://github.com/HappyCerberus/book-cpp-algorithms) - Šimon Tóth (PDF, LaTeX) *(:construction: in process)*
+* [A Complete Guide to Standard C++ Algorithms](https://github.com/HappyCerberus/book-cpp-algorithms) - Šimon Tóth (PDF, LaTeX)(CC BY-NC-SA) *(:construction: in process)*
```
Comment (eshellman) [inline books/free-programming-books-langs.md:515]:
```suggestion
* [A Complete Guide to Standard C++ Algorithms](https://github.com/HappyCerberus/book-cpp-algorithms) - Šimon Tóth (PDF, LaTeX) (CC BY-NC-SA) *(:construction: in process)*
```

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -512,15 +512,15 @@ Books on general-purpose programming that don't focus on a specific language are
...
 * [An Introduction to the USA Computing Olympiad, C++ Edition](https://darrenyao.com/usacobook/cpp.pdf) - Darren Yao (PDF)
 * [C++ Annotations](https://fbb-git.gitlab.io/cppannotations/) - Frank B. Brokken (HTML, PDF)
 * [C++ Coding Standard](https://possibility.com/Cpp/CppCodingStandard.html) - Todd Hoff (HTML, PDF)
 * [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md) - `edt.:` Bjarne Stroustrup, Herb Sutter
 * [C++ For Programmers](https://tfetimes.com/wp-content/uploads/2015/04/c-for-c-programmers.pdf) - JT Kalnay (PDF)
-* [C++ GUI Programming With Qt 3](https://ptgmedia.pearsoncmg.com/images/0131240722/downloads/blanchette_book.pdf) - Jasmin Blanchette, Mark Summerfield (PDF)
-* [C++ Language](http://www.cplusplus.com/doc/tutorial/)
-* [C++ Notes for Professionals](https://goalkicker.com/CPlusPlusBook) - Compiled from StackOverflow Documentation (PDF)
+* [C++ GUI Programming With Qt 3](https://ptgmedia.pearsoncmg.com/images/0131240722/downloads/blanchette_book.pdf) - Jasmin Blanchette, Mark Summerfield (PDF)(OPL)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:521]: OPL is deprecated. We are not supporting it by adding license annotations.

**Review by eshellman — CHANGES_REQUESTED** [review]
the extra space shoul satisfy the linter

**Review by manvith12 — COMMENTED** [review]
made the changes asked for

File: books/free-programming-books-langs.md [hunk 3]
```diff
@@ -512,15 +512,15 @@ Books on general-purpose programming that don't focus on a specific language are
...
 * [An Introduction to the USA Computing Olympiad, C++ Edition](https://darrenyao.com/usacobook/cpp.pdf) - Darren Yao (PDF)
 * [C++ Annotations](https://fbb-git.gitlab.io/cppannotations/) - Frank B. Brokken (HTML, PDF)
 * [C++ Coding Standard](https://possibility.com/Cpp/CppCodingStandard.html) - Todd Hoff (HTML, PDF)
 * [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md) - `edt.:` Bjarne Stroustrup, Herb Sutter
 * [C++ For Programmers](https://tfetimes.com/wp-content/uploads/2015/04/c-for-c-programmers.pdf) - JT Kalnay (PDF)
 * [C++ GUI Programming With Qt 3](https://ptgmedia.pearsoncmg.com/images/0131240722/downloads/blanchette_book.pdf) - Jasmin Blanchette, Mark Summerfield (PDF)
-* [C++ Language](http://www.cplusplus.com/doc/tutorial/)
-* [C++ Notes for Professionals](https://goalkicker.com/CPlusPlusBook) - Compiled from StackOverflow Documentation (PDF)
+* [C++ Language](http://www.cplusplus.com/doc/tutorial/) (HTML)
+* [C++ Notes for Professionals] (https://goalkicker.com/CPlusPlusBook) - Compiled from StackOverflow Documentation (PDF) (CC BY-SA)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:523]:
```suggestion
* [C++ Notes for Professionals](https://goalkicker.com/CPlusPlusBook) - Compiled from StackOverflow Documentation (PDF) (CC BY-SA)
```

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**